src/main/java/seedu/address/logic/commands/ImportCommand.java

```diff
@@ -0,0 +1,178 @@
+            "Remark command not implemented yet";
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Imports contacts from specified filepath "
+            + "Existing contacts will.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + "[" + PREFIX_IMPORT + "import] "
+            + "Example: " + COMMAND_WORD + PREFIX_IMPORT + "C:usr/lib/text.csv";
+    private static final String MESSAGE_ARGUMENTS = "filePath: %s";
+    private static final String MESSAGE_IMPORT_SUCCESS = "Imported Contacts from: %s";
+    private static final String MESSAGE_DATA_LOAD_ERROR = "Unable to load data from %s";
+    private static final String MESSAGE_PARSE_ERROR = "Invalid data format in %s";
+    private final Path filePath;
+    private final AddCommandParser addCommandParser = new AddCommandParser();
+
+    /**
+     * Represents the order of the data that should be parsed into the addCommandParser
+     */
+    private final String[] header = {"name", "phone", "email", "address", "matric", "reflection", "studio", "tags"};
+
+    /**
+     * Represents a mapping of String to prefix of the data that should be parsed into the addCommandParser.
+     */
+    private final Map<String, Prefix> prefixMap = Map.of(
+            "name", PREFIX_NAME,
+            "phone", PREFIX_PHONE,
+            "email", PREFIX_EMAIL,
+            "address", PREFIX_ADDRESS,
+            "matric", PREFIX_MATRIC_NUMBER,
+            "reflection", PREFIX_REFLECTION,
+            "studio", PREFIX_STUDIO,
+            "tags", PREFIX_TAG
+    );
+
+    /**
+     * @param filePath absolute path of file (path starts from C:...)
+     */
+    public ImportCommand(Path filePath) {
+        requireNonNull(filePath);
+
+        this.filePath = filePath;
+    }
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        try {
+            List<Map<String, String>> data = readCsvFile();
+            for (Map<String, String> personData : data) {
+                try {
+                    String addCommandInput = convertToAddCommandInput(personData);
+                    AddCommand addCommand = parseAddCommand(addCommandInput);
+                    addCommand.execute(model);
+                } catch (ParseException e) {
+                    throw new CommandException(String.format(MESSAGE_PARSE_ERROR, personData));
+                }
+            }
+        } catch (DataLoadingException e) {
+            throw new CommandException(String.format(MESSAGE_DATA_LOAD_ERROR, filePath));
+        }
+
+        return new CommandResult(String.format(MESSAGE_IMPORT_SUCCESS, filePath.toString()));
+    }
+    /**
+     * Reads the csv file and returns a list of maps,
+     * where each map represents a row of person's data in the csv file.
+     * @throws DataLoadingException
+     */
+    public List<Map<String, String>> readCsvFile() throws DataLoadingException {
```

Comment:
I really like the SLAP here! The cleanliness of this method keeps nested loops down to a minimum.